### PR TITLE
Fixes #26005 - track tasks on repo refresh

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by ehelms** [comment]
[test katello]

File: app/services/katello/pulp/repository.rb [hunk 1]
```diff
@@ -149,29 +149,32 @@ def proxy_host_importer_value
       end
 
       def refresh
-        update_or_associate_importer
+        tasks = update_or_associate_importer
         update_or_associate_distributors
         remove_unnecessary_distributors
+        tasks
```
Comment (sjha4) [inline app/services/katello/pulp/repository.rb]: Do we not want to track distributor tasks if we are doing so for importer tasks?

Comment (jlsherrill) [reply app/services/katello/pulp/repository.rb]: I based this on older code which didn't seem to indicate that distributor associate/delete/update tasks were async, but i guess i should double check that :)

Comment (jlsherrill) [reply app/services/katello/pulp/repository.rb]: turns out some (but not all) of those distributor tasks do generate tasks, added additional tracking

**Comment by jlsherrill** [comment]
[test katello]

**Pull request merged**